plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -319,7 +348,53 @@ private void doVacuum(
                     transactionLogFiles,
                     retainedKnownFiles,
                     retainedUnknownFiles,
-                    removedFiles);
+                    totalFilesToDelete);
+        }
+    }
+
+    private void logVacuumStart(DeltaLakeTableHandle handle, ConnectorSession session, long readVersion, long commitVersion, long numFilesToDelete, long filesToDeleteSize)
+            throws IOException
+    {
+        if (!deltaLakeConfig.isVacuumLoggingEnabled()) {
+            return;
+        }
+
+        long createdTime = System.currentTimeMillis();
+
+        TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, handle.location());
+        transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(commitVersion, createdTime, session, "VACUUM START", ImmutableMap.of("queryId", session.getQueryId(), "numFilesToDelete", String.valueOf(numFilesToDelete), "sizeOfDataToDelete", String.valueOf(filesToDeleteSize)), readVersion));
+
+        transactionLogWriter.flush();
+    }
+
+    private void logVacuumEnd(DeltaLakeTableHandle handle, ConnectorSession session, long readVersion, long commitVersion, int numDeletedFiles, int numVacuumedDirectories, String status)
+            throws IOException
+    {
+        if (!deltaLakeConfig.isVacuumLoggingEnabled()) {
+            return;
         }
+
+        long createdTime = System.currentTimeMillis();
+
+        TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, handle.location());
+        transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(commitVersion, createdTime, session, "VACUUM END", ImmutableMap.of("queryId", session.getQueryId(), "numDeletedFiles", String.valueOf(numDeletedFiles), "numVacuumedDirectories", String.valueOf(numVacuumedDirectories), "status", status), readVersion)); // TODO: blindAppend false OK?
+        transactionLogWriter.flush();
+    }
+
+    private CommitInfoEntry getCommitInfoEntry(long commitVersion, long createdTime, ConnectorSession session, String operation, ImmutableMap<String, String> metrics, long readVersion)
+    {
+        return new CommitInfoEntry(
+                commitVersion,
+                createdTime,
+                session.getUser(),
+                session.getUser(),
+                operation,
+                metrics,
```

Comment:
We currently only support operation parameters. operation metrics is another field in the commit info entry.

<img width="2194" alt="image" src="https://github.com/user-attachments/assets/c17af3e6-fb8d-4e86-9e71-2a538b3f80fa">

Reply:
yes, there is a todo in the history table class to eventually fill those columns as well.

Reply:
I added operationMetrics in `CommitInfoEntry`, I should probably prepend a commit for it.

Reply:
Now history table supports both fields